clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -358,12 +364,24 @@ func (w *waitForControlledPodsRunningMeasurement) updateOpResourceVersion(runtim
 }
 
 // getObjectCountAndMaxVersion returns number of objects that satisfy measurements parameters
-// and maximal resource version of these objects.
-// These two values allow to properly handle all object operations:
-// - When create/delete operation are called we expect the exact number of objects.
-// - When objects is updated we expect to receive event referencing this specific version.
-//   Using maximum from objects resource versions assures that all updates will be processed.
-func (w *waitForControlledPodsRunningMeasurement) getObjectCountAndMaxVersion() (int, uint64, error) {
+// and upper bound for maximal resource version of these objects.
+//
+// We can't guarantee exact value fo maximal resource version, as deletions are problematic.
+// When the object is deleted, by definition it will not be returned by the LIST request,
+// and thus we cannot infer resource version of the deletion itself.
+// To solve this problem, we:
+// - take maximum from resource versions of returned objects - this solves the creations
+//   updates case
+// - wait until set of objects in local cache is equal to the one returned from LIST call
+//   Note, that we can't compare whole objects, as in the meantime local cache may have
+//   gone ahead of the original LIST (e.g. due to status updates of existing objects).
+// - equal set of objects guarantees that all deletions are already reflected in our local
```

Comment:
This doesn't look like the other steps, i.e.

1. take maximum rv..
2. wait until...
4. return ...

Shouldn't this be part of step 2?

Reply:
done